openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java

```diff
@@ -787,13 +788,28 @@ protected Schema resolveSchema(OpenAPI openAPI, @Nullable Element definingElemen
             }
         }
 
-        ClassElement componentType = type.getFirstTypeArgument().orElse(null);
-        Map<String, ClassElement> typeArgs = type.getTypeArguments();
+        Boolean isArray = null;
+        Boolean isIterable = null;
+
+        ClassElement componentType = type != null ? type.getFirstTypeArgument().orElse(null) : null;
```

Comment:
Don't agree. Without this changes this test will be failed:
```java
    @Get("/get1")
    public <T extends Channel> String index1(@Nullable @QueryValue T[] channels) {
        return null;
    }
```

Reply:
Why exactly? It will not return that it’s an array?

Reply:
Because I'm expecting to see an EnumElement or an element, but I get `GenericPlacehoderElement`, so the processing goes the wrong way. If I comment this code, problem will be in red line:
![изображение](https://github.com/micronaut-projects/micronaut-openapi/assets/48474279/4b8b3376-37c1-41d4-bc2e-894709da745a)
I suspect that the problem arises with enums

Reply:
Right

Reply:
I think we should have something like asEnum(), we have to complicated hierarchy

Reply:
Well, but I think at the moment this fix is quite correct and should be merged

Reply:
Sure